src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala

```diff
@@ -27,15 +27,15 @@ class EchoService extends BusinessService with BusinessRequestValidation {
 
   override def validateRequest(request: BusinessServiceRequest): ValidationResult =
     Validated.cond(
-      request.text.trim.nonEmpty,
+      request.text.nonEmpty,
       request,
       NonEmptyChain.one(TextIsNonEmptyCharacters("Echo Text cannot be empty."))
     )
 
   override def process(request: BusinessServiceRequest): Either[BusinessServiceError, BusinessServiceReply] =
-    if (request.text.contains("ERROR")) {
-      Left(EchoExplicitError(s"Explicit error: ${request.text}"))
+    if (request.text.get.contains("ERROR")) {
```

Comment:
Using `get` could throw a `NoSuchElementException` if it is `None`. In this case you don't need to change anying as `contains` [can be used on an Option](https://www.scala-lang.org/api/current/scala/Option.html#contains[A1%3E:A](elem:A1):Boolean).

Reply:
Added `nonEmpty` check